backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/snapshots/SnapshotsManager.java

```diff
@@ -323,7 +326,9 @@ private String generateVmConfiguration(VM vm, List<DiskImage> disks, Map<Guid, V
         if (vm.getStaticData().getVmInit() == null) {
             vmHandler.updateVmInitFromDB(vm.getStaticData(), true);
         }
-        vmHandler.updateNumaNodesFromDb(vm);
+        if (snaphostType != SnapshotType.NEXT_RUN) {
```

Comment:
how about doing it in a similar way to how it's done for vm-init above, i.e., checking if numa nodes are already set in the vm object and load it from the db only if missing? so if there is (or would be) a flow in which it is already loaded, we won't need to go to the database..